packages/node_modules/@webex/plugin-meetings/src/statsAnalyzer/index.js

```diff
@@ -0,0 +1,373 @@
+    super();
+    this.statsResults = defaultStats;
+    this.config = config;
+    this.packetLossArraySender = {audio: [], video: [], share: []};
+    this.packetLossArrayReceiver = {audio: [], video: [], share: []};
+    this.peerConnection = null;
+  }
+
+  /**
+   * Starts the stats analyzer on interval
+   *
+   * @public
+   * @memberof StatsAnalyzer
+   * @param {PeerConnection} peerConnection
+   * @returns {void}
+   */
+  startAnalyzer(peerConnection) {
+    this.peerConnection = peerConnection;
+    this.peerConnection.onconnectionstatechange = () => {
+      // Automatically stop once the connection state is closed
+      if (this.peerConnection.connectionState === PEER_CONNECTION_STATE.CLOSED) {
+        this.stopAnalyzer();
+      }
+    };
+
+    this.statsInterval = setInterval(() => {
+      this.getStatsAndParse(this.peerConnection);
+    }, this.config.analyzerInterval);
+  }
+
+  /**
+   * Cleans up the analyzer when done
+   *
+   * @public
+   * @memberof StatsAnalyzer
+   * @returns {void}
+   */
+  stopAnalyzer() {
+    if (this.statsInterval) {
+      clearInterval(this.statsInterval);
+    }
+    this.peerConnection = null;
+  }
+
+  /**
+   * Parse a single result of get stats
+   *
+   * @private
+   * @param {*} getStatsResult
+   * @param {String} type
+   * @returns {void}
+   * @memberof StatsAnalyzer
+   */
+  parseGetStatsResult(getStatsResult, type) {
+    if (!getStatsResult) {
+      return;
+    }
+
+    switch (getStatsResult.type) {
+      case 'outbound-rtp':
+        this.processOutboundRTPResult(getStatsResult, type);
+        break;
+      case 'inbound-rtp':
+        this.processInboundRTPResult(getStatsResult, type);
+        break;
+      case 'track':
+        this.processTrackResult(getStatsResult, type);
+        break;
+      case 'remote-inbound-rtp':
+      case 'remote-outbound-rtp':
+        this.compareSentAndReceived(getStatsResult, type);
+        break;
+      default:
+        break;
+    }
+  }
+
+  /**
+   * Filters the get stats results for types
+   * @private
+   * @param {Array} getStatsResults
+   * @param {String} type
+   * @returns {void}
+   */
+  filterAndParseGetStatsResults(getStatsResults, type) {
+    const {types} = DEFAULT_GET_STATS_FILTER;
+
+    getStatsResults.forEach((result) => {
+      if (types.includes(result.type)) {
+        this.parseGetStatsResult(result, type);
+      }
+    });
+  }
+
+  /**
+   * Does a `getStats` on all the transceivers and parses the results
+   *
+   * @private
+   * @memberof StatsAnalyzer
+   * @param {*} pc
+   * @returns {void}
+   */
+  getStatsAndParse(pc) {
+    LoggerProxy.logger.info('StatsAnalyzer:index->getStatsAndParse#Collecting Stats');
+    pc.videoTransceiver.sender.getStats().then((res) => {
+      this.filterAndParseGetStatsResults(res, STATS.VIDEO_CORRELATE);
+    });
+
+    pc.videoTransceiver.receiver.getStats().then((res) => {
+      this.filterAndParseGetStatsResults(res, STATS.VIDEO_CORRELATE);
+    });
```

Comment:
```suggestion
    ['sender', 'receiver'].forEach((direction) => {
      pc.videoTransceiver[direction].getStats().then((res) => {
        this.filterAndParseGetStatsResults(res, STATS.VIDEO_CORRELATE);
      })
    });
```
Since it looks like you do the exact same thing for both send and receive